### ISSUE TITLE
 unexpected 'fn' (T_STRING), expecting :: (T_PAAMAYIM_NEKUDOTAYIM) 

### ISSUE THREAD

**Issue by balrick** [kreait/firebase-php#820]
### Describe the bug

Using a Shared environment and Lumen 8.  Using composer to install however because ext-sodium is not allowed on the shared server.

Code use:  Composer install   --ignore-platform-req=ext-sodium

PHP V 7.4.33

### Installed packages

```json
{
    "name": "laravel/lumen",
    "description": "The Laravel Lumen Framework.",
    "keywords": ["framework", "laravel", "lumen"],
    "platform-check": false,
    "license": "MIT",
    "type": "project",

    "require": {
        "php": "^7.3|^8.0",
        "darkaonline/swagger-lume": "8.*",
        "doctrine/dbal": "^3.3",
        "flipbox/lumen-generator": "^8.2",
        "graham-campbell/flysystem": "^7.1",
        "kreait/firebase-php": "^6.0",
        "kreait/laravel-firebase": "^4.1",
        "laravel/lumen-framework": "^8.3.1",
        "webpatser/laravel-uuid": "2.0"
    },
    "require-dev": {
        "fakerphp/faker": "^1.9.1",
        "mockery/mockery": "^1.3.1",
        "phpunit/phpunit": "^9.5.10"
    },
    "autoload": {
        "psr-4": {
            "App\\": "app/",
            "Database\\Factories\\": "database/factories/",
            "Database\\Seeders\\": "database/seeders/"
        }
    },
    "autoload-dev": {
        "classmap": [
            "tests/"
        ]
    },
    "config": {
        "preferred-install": "dist",
        "sort-packages": true,
        "optimize-autoloader": true,
        "platform-check":false,
        "platform": {
            "php": "7.4.33"
         }
    },
    "minimum-stability": "dev",
    "prefer-stable": true,
    "scripts": {
        "post-root-package-install": [
            "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
        ]
    }
}
```

### PHP version and extensions

```shell
composer                2.2.12   Composer package
composer-plugin-api     2.2.0    The Composer Plugin API
composer-runtime-api    2.2.2    The Composer Runtime API
ext-apcu                5.1.22   The apcu PHP extension
ext-bcmath              7.4.33   The bcmath PHP extension
ext-bz2                 7.4.33   The bz2 PHP extension
ext-calendar            7.4.33   The calendar PHP extension
ext-ctype               7.4.33   The ctype PHP extension
ext-curl                7.4.33   The curl PHP extension
ext-date                7.4.33   The date PHP extension
ext-dom                 20031129 The dom PHP extension
ext-exif                7.4.33   The exif PHP extension
ext-fileinfo            7.4.33   The fileinfo PHP extension
ext-filter              7.4.33   The filter PHP extension
ext-ftp                 7.4.33   The ftp PHP extension
ext-gd                  7.4.33   The gd PHP extension
ext-gettext             7.4.33   The gettext PHP extension
ext-gmp                 7.4.33   The gmp PHP extension
ext-hash                7.4.33   The hash PHP extension
ext-iconv               7.4.33   The iconv PHP extension
ext-imagick             3.4.3    The imagick PHP extension
ext-imap                7.4.33   The imap PHP extension
ext-intl                7.4.33   The intl PHP extension
ext-ioncube-loader      10.4.5   The ionCube Loader PHP extension
ext-json                7.4.33   The json PHP extension
ext-libxml              7.4.33   The libxml PHP extension
ext-mbstring            7.4.33   The mbstring PHP extension
ext-mysqli              7.4.33   The mysqli PHP extension
ext-mysqlnd             0        The mysqlnd PHP extension (actual version:...
ext-odbc                7.4.33   The odbc PHP extension
ext-openssl             7.4.33   The openssl PHP extension
ext-pcntl               7.4.33   The pcntl PHP extension
ext-pcre                7.4.33   The pcre PHP extension
ext-pdo                 7.4.33   The PDO PHP extension
ext-pdo_mysql           7.4.33   The pdo_mysql PHP extension
ext-pdo_odbc            7.4.33   The PDO_ODBC PHP extension
ext-pdo_pgsql           7.4.33   The pdo_pgsql PHP extension
ext-pgsql               7.4.33   The pgsql PHP extension
ext-phar                7.4.33   The Phar PHP extension
ext-posix               7.4.33   The posix PHP extension
ext-pspell              7.4.33   The pspell PHP extension
ext-readline            7.4.33   The readline PHP extension
ext-reflection          7.4.33   The Reflection PHP extension
ext-session             7.4.33   The session PHP extension
ext-simplexml           7.4.33   The SimpleXML PHP extension
ext-soap                7.4.33   The soap PHP extension
ext-sockets             7.4.33   The sockets PHP extension
ext-sourceguardian      12.1.2   The SourceGuardian PHP extension
ext-spl                 7.4.33   The SPL PHP extension
ext-tidy                7.4.33   The tidy PHP extension
ext-tokenizer           7.4.33   The tokenizer PHP extension
ext-xml                 7.4.33   The xml PHP extension
ext-xmlreader           7.4.33   The xmlreader PHP extension
ext-xmlrpc              7.4.33   The xmlrpc PHP extension
ext-xmlwriter           7.4.33   The xmlwriter PHP extension
ext-xsl                 7.4.33   The xsl PHP extension
ext-zend-opcache        7.4.33   The Zend OPcache PHP extension
ext-zip                 1.15.6   The zip PHP extension
ext-zlib                7.4.33   The zlib PHP extension
lib-bz2                 1.0.5    The bz2 library
lib-curl                8.2.1    The curl library
lib-curl-libssh2        1.4.2    curl libssh2 version
lib-curl-openssl        1.1.1.21 curl OpenSSL version (1.1.1.21)
lib-curl-zlib           1.2.3    curl zlib version
lib-date-timelib        2018.04  date timelib version
lib-date-zoneinfo       0        zoneinfo ("Olson") database for date
lib-fileinfo-libmagic   537      fileinfo libmagic version
lib-gd                  2.0.35   The gd library
lib-gd-freetype         2.3.11   freetype version for gd
lib-gd-libjpeg          6.2      libjpeg version for gd
lib-gd-libpng           1.2.49   libpng version for gd
lib-gd-libxpm           3.4.11   libxpm version for gd
lib-gmp                 4.3.1    The gmp library
lib-iconv               2.12     The iconv library
lib-icu                 69.1     The ICU unicode and globalization support ...
lib-icu-cldr            39       ICU CLDR project version
lib-icu-unicode         13.0.0   ICU unicode version
lib-icu-zoneinfo        2021.1   zoneinfo ("Olson") database for icu
lib-imagick-imagemagick 6.7.2.7  The imagick-imagemagick library
lib-libxml              2.9.7    libxml library version
lib-libxslt             1.1.26   The libxslt library
lib-libxslt-libxml      2.7.6    libxml version libxslt is compiled against
lib-mbstring-libmbfl    1.3.2    mbstring libmbfl version
lib-mbstring-oniguruma  6.9.8    mbstring oniguruma version
lib-openssl             1.1.1.21 OpenSSL 1.1.1u  30 May 2023
lib-pcre                10.35    The pcre library
lib-pcre-unicode        13.0.0   PCRE Unicode version support
lib-pdo_pgsql-libpq     8.4.20   libpq for pdo_pgsql
lib-pgsql-libpq         8.4.20   libpq for pgsql
lib-zip-libzip          1.9.2    The zip-libzip library
lib-zlib                1.2.3    The zlib library
php                     7.4.33   Package overridden via config.platform, sa...
php-64bit               7.4.33   Package overridden via config.platform, sa...
php-ipv6                7.4.33   Package overridden via config.platform, sa..
```

### Steps to reproduce the issue.

```shell
# Insert the commands issued in the terminal if they are needed
# to reproduce the issue. Otherwise, delete this code block.
```

```php
// Insert the PHP code to reproduce the issue. Please ensure that it is code that
// can be copy pasted to reproduce it.
```

### Error message/Stack trace

```shell
[2023-08-03 00:01:32] local.ERROR: syntax error, unexpected 'fn' (T_STRING), expecting :: (T_PAAMAYIM_NEKUDOTAYIM) {"exception":"[object] (ParseError(code: 0): syntax error, unexpected 'fn' (T_STRING), expecting :: (T_PAAMAYIM_NEKUDOTAYIM) at /deployFolder/vendor/kreait/laravel-firebase/src/ServiceProvider.php:46)
[stacktrace]
#0 /deployFolder/vendor/composer/ClassLoader.php(428): Composer\\Autoload\\includeFile('/home2/bashadev...')
#1 [internal function]: Composer\\Autoload\\ClassLoader->loadClass('Kreait\\\\Laravel\\\\...')
#2 /deployFolder/vendor/laravel/lumen-framework/src/Application.php(223): spl_autoload_call('Kreait\\\\Laravel\\\\...')
#3 /deployFolder/bootstrap/app.php(110): Laravel\\Lumen\\Application->register('Kreait\\\\Laravel\\\\...')
#4 /home2/bashadev/alrayfx/api/index.php(14): require('/home2/bashadev...')
#5 {main}
"}
[2023-08-03 00:16:32] local.ERROR: syntax error, unexpected 'fn' (T_STRING), expecting :: (T_PAAMAYIM_NEKUDOTAYIM) {"exception":"[object] (ParseError(code: 0): syntax error, unexpected 'fn' (T_STRING), expecting :: (T_PAAMAYIM_NEKUDOTAYIM) at /deployFolder/vendor/kreait/laravel-firebase/src/ServiceProvider.php:46)
```

**Comment by jeromegamez** [comment]
This version of the SDK is not supported anymore (see https://github.com/kreait/firebase-php#supported-versions).

Please update to the current version, or see https://github.com/kreait/firebase-php#support-for-older-versions-of-the-sdk if you would like me to look into this issue.

**Comment by balrick** [comment]
I have tried multiple lower version and still get  same error.  Which version do you recommend to use for php 7.4 and Lumen 8

Sent from my iPhone

On Aug 3, 2023, at 5:15 PM, Jérôme Gamez ***@***.***> wrote:

﻿

Closed #820<https://github.com/kreait/firebase-php/issues/820> as completed.
